src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -85,6 +87,20 @@ public void addMessage(Message message) {
     return messagesInConversation;
   }
 
+  /** Access the set of Messages sent by the user. */
+  public List<Message> getMessagesByUser(UUID author) {
+
+    List<Message> messagesByUser = new ArrayList<>();
+
+    for (Message message : messages) {
+      if (message.getAuthorId().equals(author)) {
+        messagesByUser.add(message);
+      }
+    }
```

Comment:
This can all be done in one line with Java 8:

```java
return messages.stream().filter(m -> m.getAuthorId().equals(author)).collect(Collectors.toList());
```